common/lib/cache-function-ttl.ts

```diff
@@ -0,0 +1,16 @@
+const cache = new Map< () => Promise< unknown >, { result: unknown; timestamp: number } >();
+
+export function cacheFunctionTTL< Args extends unknown[], Return >(
+	fn: ( ...args: Args ) => Promise< Return >,
+	ttl = 1 * 1000
+) {
+	return async ( ...args: Args ) => {
+		const cachedValue = cache.get( fn );
+		if ( cachedValue && Date.now() - cachedValue.timestamp < ttl ) {
+			return cachedValue.result as Return;
+		}
+		const value = await fn( ...args );
+		cache.set( fn, { result: value, timestamp: Date.now() } );
+		return value;
+	};
+}
```

Comment:
```suggestion
export function cacheFunctionTTL< Args extends unknown[], Return >(
	fn: ( ...args: Args ) => Promise< Return >,
	ttl = 1000
) {
	const cache = new Map< string, { result: Return; timestamp: number } >();

	return async ( ...args: Args ) => {
		const key = JSON.stringify( args );
		const cachedValue = cache.get( key );
		if ( cachedValue && Date.now() - cachedValue.timestamp < ttl ) {
			return cachedValue.result;
		}
		const value = await fn( ...args );
		cache.set( key, { result: value, timestamp: Date.now() } );
		return value;
	};
}
```

Currently this function has a bug for calls with different arguments.
A test like this would fail:

```typescript
it( 'should return different results for different arguments', async () => {
		const fetchUser = jest.fn( async ( userId: number ) => {
			return { id: userId, name: `User ${ userId }` };
		} );

		const cachedFetchUser = cacheFunctionTTL( fetchUser, 1000 );

		const user1 = await cachedFetchUser( 1 );
		const user2 = await cachedFetchUser( 2 );

		expect( user1 ).toEqual( { id: 1, name: 'User 1' } );
		expect( user2 ).toEqual( { id: 2, name: 'User 2' } );
	} );
```

Reply:
Good catch 👍 I've updated `cacheFunctionTTL` to account for function arguments, too (using `fast-deep-equal` to compare function arguments)